pylint/plugins/hass_enforce_type_hints.py

```diff
@@ -485,19 +556,52 @@ class HassTypeHintChecker(BaseChecker):  # type: ignore[misc]
     def __init__(self, linter: PyLinter | None = None) -> None:
         super().__init__(linter)
         self._function_matchers: list[TypeHintMatch] = []
+        self._class_matchers: list[ClassTypeHintMatch] = []
 
     def visit_module(self, node: astroid.Module) -> None:
         """Called when a Module node is visited."""
         self._function_matchers = []
+        self._class_matchers = []
 
         if (module_platform := _get_module_platform(node.name)) is None:
             return
 
         if module_platform in _PLATFORMS:
             self._function_matchers.extend(_FUNCTION_MATCH["__any_platform__"])
 
-        if matches := _FUNCTION_MATCH.get(module_platform):
-            self._function_matchers.extend(matches)
+        if function_matches := _FUNCTION_MATCH.get(module_platform):
+            self._function_matchers.extend(function_matches)
+
+        if class_matches := _CLASS_MATCH.get(module_platform):
+            self._class_matchers = class_matches
+
+    def visit_classdef(self, node: astroid.ClassDef) -> None:
+        """Called when a ClassDef node is visited."""
+        for class_matches in self._class_matchers:
+            for base in node.bases:
```

Comment:
`node.bases` only iterates over the immediate base class nodes. Thus it doesn't capture nested subclasses which don't directly inherit from `ConfigFlow`. Try using `node.ancestors()`. That will iterate over all base classes until `object`.

Some notes:
* `ancestors()` uses inference. Thus is a bit more time consuming, not much though. However it might make sense to swap the `for` loops.
* Your current test cases won't work since astroid can't infer `config_entries.ConfigFlow`.